quilter.py

```diff
@@ -0,0 +1,360 @@
+'''
+quilter.py
+
+Download ALL THE RASTERS!!!
+
+'''
+
+import csv
+import os
+import tempfile
+import zipfile
+import shutil
+import sys
+import requests
+import traceback
+import inspect
+import subprocess
+import argparse
+from pathlib import Path
+
+# User Input:
+# csv file
+# destination folder
+# -m merge flag and name
+# -p project flag and accompanying EPSG:xxxxx designator
+# -d don't compress tif (ignore for aerial imagery)
+
+parser = argparse.ArgumentParser(description='Download helper for\
+                                              raster.utah.gov')
+
+parser.add_argument('csv', help='CSV file of download links.')
+parser.add_argument('destination', 
+                    help='Directory for extracted files (must already exist).')
+
+parser.add_argument('-m', '--merge', dest="name", 
+                    help='Merge downloaded files to specified base name. .tif\
+                    or .shp extensions will be added as appropriate. Requires\
+                    GDAL.')
+parser.add_argument('-p', '--project', dest="crs", help='Reproject merged file\
+                    to specified CRS. Specify CRS like EPSG:x or ESRI:x.\
+                    Requires -m.')
+parser.add_argument('-d', '--dont_compress', help='Disable LZW compression of\
+                    merged raster (all aerials will remain compressed).',
+                    action='store_true')
+
+args = parser.parse_args()
+
+# Set defaults
+merge = False
+project = False
+compress = True
+delete = False
+
+outfolder = args.destination
+csv_file = args.csv
+if args.name:
+    final_name = args.name
+    merge = True
+if args.crs:
+    projection = args.crs
+    project = True
+if args.dont_compress:
+    compress = False
+
+# outfolder = 'c:\\gis\\elevation\\statewide\\test'
+# csv_file = 'c:\\users\\jdadams\\downloads\\10mDEMS2.csv'
+# projection = "EPSG:26912"
+# final_name = "10m"
+
+ext_list = ('.img', '.tif', '.asc', '.jpg', '.jgw', '.xyz', '.tfw', '.dbf', 
+            '.prj', '.shp', '.shx')
+raster_exts = ('.img', '.tif', '.asc', '.jpg')
+
+# merge = True
+# project = True 
+# compress = True  # Default to always compress rasters w/ lzw
+
+# TODO: Need to set these automagically. Maybe something in CSV?
+raster = True
+vector = False
+
+
+# Progress bar for download
```

Comment:
we have a little style convention that we use `#: for comments` and `# for commented out code`. Typically we do not like to keep commented out code in the files. I see you are using them for testing now, but there are other ways.

Reply:
I'll switch to the `#:` convention. I agree; commented out code is not ideal in published projects. What are these other ways you speak of? Branching with git?

Reply:
It is kind of situationally dependent but we have created python modules that you can import and use values. We often do this for app secrets e.g., [sample](https://github.com/steveoh/bulldozer/blob/master/servers.sample.py). Then your testing or production secrets can be ignored from git yet you can still use them. 

Other ways would be to create cli options or flags to set some of the optional things.